### PR TITLE
Test GaussianMI with `normalized=false` and `true`

### LINKED ISSUE #318: Test for GaussianMI
I'm not super well versed in the theory, but I can try to come up with a test that compares, say, the `MI(normalize=true)` of unnormalized data with the `MI(normalize=false)` of the already normalized data, and then also compare those to the theoretical result for some random samples (with fixed seed). Does that seem reasonable?a

_Originally posted by @RomeoV in https://github.com/JuliaDynamics/CausalityTools.jl/issues/316#issuecomment-1690607237_

### PR DESCRIPTION
**Pull request by RomeoV**
Closes #318 once tests pass.

- [x] include both normalize=true and false in regular testing
- [x] make sure both versions are consistent
- [x] compare with analytic eq

~Test currently fails, although I believe the eq is correct. Probably a bug in the math or implementation somewhere?~

### CODE REVIEW

**Comment by kahaaga** [comment]
Thanks for the contribution, @RomeoV! 

The tests will fail on this branch because the only file that has changed in this PR relative to  the `main` is the test file you modified. Therefore, the bug that you found for `GaussianMI` is still present in this `PR`. 

You can try merging your fork of the `main` branch onto this branch (the `test-gaussian-mi` branch). Then, if everything is correct after your changes in #318, the tests should hopefully pass here.

**Comment by kahaaga** [comment]
Hey again, @RomeoV. After you merge your changes on `master` onto this branch, some tests will still fail. However, these are unrelated to the MI tests you added here (see #320). 

Take-away: When the MI tests you added pass, this PR is good. The other errors are due to some package requirements - I'll have a look at fixing those, and will release a new patch version of the package, once this PR is good to go.

**Comment by RomeoV** [comment]
Test should pass now. I noticed that the default base of `MIShannon` is 2, but it's `e` "for the math". So I fixed that and had to increase the sample size a bit, and now it looks good!

Sidenote: I suppose `base=2` is common for computer science information theory, but this problem took me a while to find, as all the math on wikipedia assumes `base=\euler`. I don't know enough about the literature to judge which default choice has "less surprise" (😉) here though.

**Review by kahaaga — CHANGES_REQUESTED** [review]
The test suite for the MI still fails due to some syntax errors.

If you didn't already know, you can run the test suite locally in the REPL by activating the `CausalityTools` project folder, then typing `]` to enter package mode, and then typing `test`. Then the test suite will run, and you will get a summary of which test sets were successful and which were not. You'll see that in the current state of this PR, there's one mutual info testset that fails (it shows up in red):
Screenshot 2023-08-25 at 10 06 26

If you fix the syntax errors I highlighted, these errors should hopefully disappear!

**Comment by RomeoV** [comment]
Sorry for that. I was a bit in a hurry, so I didn't want to run the whole test harness (although I should have) and just ran the inner code in a REPL. Thanks for picking up on it. Should be fixed now.

**Comment by kahaaga** [comment]
> Sorry for that. I was a bit in a hurry, so I didn't want to run the whole test harness (although I should have) and just ran the inner code in a REPL. Thanks for picking up on it. Should be fixed now.

Perfect! Thank you so much for your effort. This looks good to go.

PS: in the future, if you contribute to this or other repos, it is helpful if you tick the "allow repo maintainers to make changes to the PR" button (I can't remember the exact wording, but there's an option like that when you open the PR).  That way, I can immediately fix minor things like the missing `begin` keywords myself, without us having to go back and forth over multiple days and time zones for minor details.

**Pull request merged**
